robot_smach_states/examples/Lethal_route_planning.py

```diff
@@ -0,0 +1,81 @@
+# ROS
+import PyKDL as kdl
+import rospy
+import math
+from std_msgs.msg import Bool
+from geometry_msgs.msg import Twist
+from nav_msgs.msg import OccupancyGrid
+
+# TU/e Robotics
+from robot_skills.get_robot import get_robot
+
+
+class Lethal_Zone:
+    def __init__(self, robot):
+        self.robot = robot
+        self.front_bumper_sub = rospy.Subscriber("hero/base_f_bumper_sensor", Bool, self.front_callback)
+        self.front_bumper_active = False
+        self.back_bumper_sub = rospy.Subscriber("hero/base_b_bumper_sensor", Bool, self.back_callback)
+        self.back_bumper_active = False
+        self._cmd_vel = rospy.Publisher("hero/base/references", Twist)
+        self.global_costmap_sub = rospy.Subscriber("/hero/global_planner/global_costmap/costmap", OccupancyGrid, self.global_costmap_callback)
+        self.costmap_info = None
+        self.costmap_data = None
+
+    def front_callback(self, msg):
+        rospy.loginfo(msg)
+        self.front_bumper_active = msg.data
+
+    def back_callback(self, msg):
+        rospy.loginfo(msg)
+        self.back_bumper_active = msg.data
+
+    def global_costmap_callback(self, msg):
+        rospy.loginfo(msg)
+        self.costmap_info = msg.info
+        self.costmap_data = msg.data
+
+    def start_value(self, x, y):
+        x = (x - self.costmap_info.origin.position.x) / self.costmap_info.resolution
+        y = (y - self.costmap_info.origin.position.y) / self.costmap_info.resolution
+        start = self.costmap_data[x, y]
+        return start
+
+    def free_space_finder(self, x, y):
+        d_max = (x+4)^2 + (y+4)^2
+        x_free = None
+        y_free = None
+        for i in range(x-3, x+4):
+            for j in range(y-3, y+4):
+                if self.costmap_data[i, j] < 253:
+                    d = (i-x) ^ 2 + (j-y) ^ 2
```

Comment:
Let op dat d de afstand in het kwadraat is in grid indices. buiten dit script gebruik je het namelijk als afstand in meters. Ik zou de afstand niet uit deze functie halen, maar later berekenen aan de hand van je begin en eindpositie in meters.